File: server/java/src/main/java/com/onlinepayments/example/service/domain/paymentdetails/PaymentProduct840CustomerAccount.java
package com.onlinepayments.example.service.domain.paymentdetails;

public record PaymentProduct840CustomerAccount(
        String accountId,
        String companyName,
        String countryCode,
        String customerAccountStatus,
        String customerAddressStatus,
        String firstName,
        String payerId,
        String surname
) { }

File: server/python/app/application/domain/payments/payment_details/payment_product_840_customer_account.py
from dataclasses import dataclass
from typing import Optional

@dataclass
class PaymentProduct840CustomerAccount:
    account_id: Optional[str] = None
    company_name: Optional[str] = None
    country_code: Optional[str] = None
    customer_account_status: Optional[str] = None
    customer_address_status: Optional[str] = None
    first_name: Optional[str] = None
    payer_id: Optional[str] = None
    surname: Optional[str] = None